test/StoragelessSessionTest/Http/SessionMiddlewareTest.php

```diff
@@ -197,9 +194,34 @@ public function testSkipsInjectingSessionCookieOnEmptyContainer(SessionMiddlewar
         self::assertNull(FigResponseCookies::get($response, SessionMiddleware::DEFAULT_COOKIE)->getValue());
     }
 
-    public function testRemovesSessionCookieOnEmptySessionContainer()
+    /**
+     * @dataProvider validMiddlewaresProvider
+     */
+    public function testRemovesSessionCookieOnEmptySessionContainer(SessionMiddleware $middleware)
     {
-        self::markTestIncomplete('This feature is yet to be implemented, and we may do so in a different middleware');
+        $validTokenRequest = (new ServerRequest())
```

Comment:
I think that things are getting messy in ```SessionMiddlewareTest```. I'll refactor it in a separated PR and we can discuss things and then rebase this branch.